android/libraries/rib-base/src/main/java/com/badoo/ribs/core/routing/configuration/feature/ConfigurationFeatureActor.kt

```diff
@@ -165,53 +153,61 @@ internal class ConfigurationFeatureActor<C : Parcelable>(
     private fun createDefaultElements(
         state: WorkingState<C>,
         commands: List<ConfigurationCommand<C>>
-    ): Map<ConfigurationKey, ConfigurationContext<C>> {
-        val defaultElements: MutableMap<ConfigurationKey, ConfigurationContext<C>> = mutableMapOf()
+    ): Pool<C> {
+        val defaultElements: MutablePool<C> = mutablePoolOf()
 
         commands.forEach { command ->
-            // TODO unify this with resolution for all other types if possible
             if (command is ConfigurationCommand.Add<C> && !state.pool.containsKey(command.key) && !defaultElements.containsKey(command.key)) {
                 defaultElements[command.key] = ConfigurationContext.Unresolved(
-                    ConfigurationContext.ActivationState.INACTIVE,
-                    command.configuration
-                ).resolve(configurationResolver, parentNode) {
-                    val action = AddAction(it, parentNode)
-                    action.onTransition()
-                    action.result
-                }
+                    activationState = ConfigurationContext.ActivationState.INACTIVE,
+                    configuration = command.key.configuration
+                )
             }
         }
 
         return defaultElements
     }
 
     private fun createParams(
+        emitter: EffectEmitter<C>,
         state: WorkingState<C>,
-        defaultElements: Map<ConfigurationKey, ConfigurationContext<C>>,
-        command: Transaction<C>? = null
-    ): ActionExecutionParams<C> =
-        ActionExecutionParams(
-            resolver = { key -> configurationKeyResolver.resolve(state, key, defaultElements) },
+        defaultElements: Pool<C>,
+        transaction: Transaction<C>? = null
+    ): TransactionExecutionParams<C> {
+        val tempPool: MutablePool<C> = state.pool.toMutablePool()
+
+        return TransactionExecutionParams(
+            emitter = emitter,
+            resolver = { key ->
+                val lookup = tempPool[key]
+                if (lookup is ConfigurationContext.Resolved) lookup
```

Comment:
Extract maybe?

Reply:
Tried, then reverted for increasing overall complexity how it ended up.